bin/hash_directory

```diff
@@ -7,8 +7,15 @@ if [ -z "$1" ]; then
 	exit 1
 fi
 
+# `shasum` is available on only macOS
+if command -v shasum &> /dev/null; then
+  sha_command=(shasum -a 256)
+else
+  sha_command=(sha256sum)
+fi
+
 # - Find all files in the given directory
 # - Run `sha256sum` on each file found – the `+` flag does it in parallel for a huge speed boost.
 # - Sort the files by filename for deterministic hashing
 # - Take the hash of all of the output hashes (and file paths)
-find "$DIRECTORY_PATH" -type f -exec sha256sum "{}" \+ | sort -k 2 | shasum -a 256 | cut -f1 -d " "
+find "${DIRECTORY_PATH%/}" -type f -exec "${sha_command[@]}" "{}" \+ | sort -k 2 | "${sha_command[@]}" | cut -f1 -d " "
```

Comment:
PS: I considered having the `find` command below calling our `hash_file` helper instead of replicating the `if…else…fi` in `hash_directory` here… but it didn't work because our `hash_file` helper requires a parameter to be passed (`$1`), and thus is not currently designed to accept taking input from STDIN.
And as we are using it that way in the second part of this command (`sort -k 2 | hash_file | cut -f1 -d " "`) it would end up returning `You` (i.e. the first word—because of `cut -f1`—of the "You must pass a filename to hash" error message) instead of the expected output 😅 

I figured that allowing `hash_file` to support `STDIN` as input just to fix this was not in scope of this PR and that it was simpler to just replicate the `if …` test here instead.